hcn/hcnnetwork_test.go

```diff
@@ -66,3 +66,61 @@ func TestListNetwork(t *testing.T) {
 		t.Error(err)
 	}
 }
+
+func TestAddPolicyNetwork(t *testing.T) {
+
+	network, err := CreateTestOverlayNetwork()
+	if err != nil {
+		t.Error(err)
+	}
+
+	remoteSubnetRoutePolicy, err := HcnCreateRemoteSubnetRoute()
+	if err != nil {
+		t.Error(err)
+	}
+
+	//Add Policy
+	network.AddPolicy(*remoteSubnetRoutePolicy)
+
+	//Reload the network object from HNS.
+	network, err = GetNetworkByID(network.Id)
```

Comment:
I'm not fond of this pattern.  We shouldn't have to reload the network after modifying.  After updating the network (or endpoint) the network should be automatically updated.  However, this behavior is consistent with the Endpoint  API's.
I'd prefer we change this behavior in a separate PR.